### ISSUE TITLE
Can not recover the result reported in the paper

### ISSUE THREAD

**Issue by XinnuoXu** [pengbaolin/SC-GPT#18]
Hi Baolin, nice work!

Here are the results I recovered, which are a bit different from the ones reported in the paper. The models are fine-tuned on scgpt. I guess they are the output of SC-GPT?

Restaurant: BLEU: 0.2986831023360097, ERR:4.491017964071856
Hotel: BLEU: 0.35690465390989745, ERR:3.8461538461538463
Laptop: BLEU: 0.2960457124834261, ERR:4.897631473303894
Tv: BLEU: 0.26934365138564614, ERR:7.526444263628966

Here are hyper-parameters I used:

++++++
Training
++++++

EPOCH=5
DOMAIN=restaurant
LR=5e-5
PRE_TRAINED_MODEL_PATH=./scgpt
MODEL_SAVE_PATH=./models.${DOMAIN}/

python train.py \
	--output_dir=${MODEL_SAVE_PATH} \
	--model_type=gpt2 \
	--model_name_or_path=${PRE_TRAINED_MODEL_PATH} \
	--do_train \
	--do_eval \
	--eval_data_file=data/${DOMAIN}/train.txt \
	--per_gpu_train_batch_size 1 \
	--num_train_epochs ${EPOCH} \
	--learning_rate ${LR} \
	--overwrite_cache \
	--use_tokenize \
	--train_data_file=data/${DOMAIN}/train.txt \
	--overwrite_output_dir

++++++
Testing
++++++

DOMAIN=restaurant
MODEL_SAVE_PATH=./models.${DOMAIN}/

python generate.py \
	--model_type=gpt2 \
	--model_name_or_path=${MODEL_SAVE_PATH} \
	--num_samples 5 \
	--input_file=data/${DOMAIN}/test.txt \
	--top_k 5 \
	--output_file=results_${DOMAIN}.json \
	--length 80

++++++
Evaluation
++++++

DOMAIN=restaurant
python evaluator.py --domain ${DOMAIN} --target_file results_${DOMAIN}.json

**Comment by pengbaolin** [comment]
Thanks!  Suggest use epoch as 10 to 20 and reduce learning rate to 1e-5. For the decoding, using top_p from 0.5 to 0.9 instead of top_k generally yields better performance. 

Below are my results .
python evaluator.py --domain hotel --target_file hotelresults.json

BLEU SCORE & SLOT ERROR on GENERATED SENTENCES

Metric       :  BLEU    T.ERR   A.ERR
HDC          :  1.0000  0.00%   0.00%
Ref          :  1.0000  0.55%   0.00%

This Model   :  0.4008  2.75%   1.65%
FIELNAME: hotelresults.json, BLEU: 0.4007618128692502, ERR:2.7472527472527473

**Comment by XinnuoXu** [comment]
The results are better now. Many thanks Baolin : )
